contracts/utils/contracts/src/v08/LibBytesV08.sol

```diff
@@ -0,0 +1,440 @@
+                )
+            );
+        }
+
+        // Create a new bytes structure around [from, to) in-place.
+        assembly {
+            result := add(b, from)
+            mstore(result, sub(to, from))
+        }
+        return result;
+    }
+
+    /// @dev Pops the last byte off of a byte array by modifying its length.
+    /// @param b Byte array that will be modified.
+    /// @return result The byte that was popped off.
+    function popLastByte(bytes memory b) internal pure returns (bytes1 result) {
+        if (b.length == 0) {
+            LibRichErrorsV08.rrevert(
+                LibBytesRichErrorsV08.InvalidByteOperationError(
+                    LibBytesRichErrorsV08.InvalidByteOperationErrorCodes.LengthGreaterThanZeroRequired,
+                    b.length,
+                    0
+                )
+            );
+        }
+
+        // Store last byte.
+        result = b[b.length - 1];
+
+        assembly {
+            // Decrement length of byte array.
+            let newLen := sub(mload(b), 1)
+            mstore(b, newLen)
+        }
+        return result;
+    }
+
+    /// @dev Tests equality of two byte arrays.
+    /// @param lhs First byte array to compare.
+    /// @param rhs Second byte array to compare.
+    /// @return equal True if arrays are the same. False otherwise.
+    function equals(bytes memory lhs, bytes memory rhs) internal pure returns (bool equal) {
+        // Keccak gas cost is 30 + numWords * 6. This is a cheap way to compare.
+        // We early exit on unequal lengths, but keccak would also correctly
+        // handle this.
+        return lhs.length == rhs.length && keccak256(lhs) == keccak256(rhs);
+    }
+
+    /// @dev Reads an address from a position in a byte array.
+    /// @param b Byte array containing an address.
+    /// @param index Index in byte array of address.
+    /// @return result address from byte array.
+    function readAddress(bytes memory b, uint256 index) internal pure returns (address result) {
+        if (b.length < index + 20) {
+            LibRichErrorsV08.rrevert(
+                LibBytesRichErrorsV08.InvalidByteOperationError(
+                    LibBytesRichErrorsV08.InvalidByteOperationErrorCodes.LengthGreaterThanOrEqualsTwentyRequired,
+                    b.length,
+                    index + 20 // 20 is length of address
+                )
+            );
+        }
+
+        // Add offset to index:
+        // 1. Arrays are prefixed by 32-byte length parameter (add 32 to index)
+        // 2. Account for size difference between address length and 32-byte storage word (subtract 12 from index)
+        index += 20;
+
+        // Read address from array memory
+        assembly {
+            // 1. Add index to address of bytes array
+            // 2. Load 32-byte word from memory
+            // 3. Apply 20-byte mask to obtain address
+            result := and(mload(add(b, index)), 0xffffffffffffffffffffffffffffffffffffffff)
```

Comment:
Masking is redundant. Solc does this for you automatically when exiting an assembly block